bootstrap/bootstrap.go

```diff
@@ -1045,8 +1045,15 @@ func (b *Bootstrap) updateGitMirror() (string, error) {
 	}
 
 	// Update our mirror
-	if err := b.shell.Run("git", "--git-dir", mirrorDir, "remote", "update", "--prune"); err != nil {
-		return "", err
+	if b.GitMirrorsFetchOnly {
+		gitFetchFlags := fmt.Sprintf("--git-dir '%s' %s", mirrorDir, b.GitFetchFlags)
+		if err := gitFetch(b.shell, gitFetchFlags, "origin", b.Branch); err != nil {
```

Comment:
I wonder if this has been tested (i.e. executed at all).
AFAIK `--git-dir` must come before the git subcommand like `fetch`, otherwise it's silently ignored.
Testing with a random git repo I have locally:

```
pda@paulbookpro ~/tmp ❯ git fetch --git-dir ~/code/example.git
fatal: not a git repository (or any of the parent directories): .git
```

```
pda@paulbookpro ~/tmp ❯ git --git-dir ~/code/example.git fetch
remote: Enumerating objects: 81, done.
remote: Counting objects: 100% (81/81), done.
remote: Compressing objects: 100% (32/32), done.
remote: Total 167 (delta 49), reused 62 (delta 40), pack-reused 86
Receiving objects: 100% (167/167), 50.87 KiB | 5.65 MiB/s, done.
Resolving deltas: 100% (81/81), completed with 16 local objects.
…
```

(I'm looking into merging #1112 which is basically a version of this PR without conditionals around the new behaviour)